PaTTAN Room Displays/PaTTAN Room Displays/App.xaml.cs

```diff
@@ -1,21 +1,94 @@
-﻿using System.Threading.Tasks;
+﻿using TimeZoneConverter;
 
 namespace PaTTAN_Room_Displays;
 
 public partial class App : Application
 {
-	RoomDataService roomDataService = new RoomDataService();
-	public static List<Meeting> meetingList = new List<Meeting>();
+	/// <summary>
+	/// Class <c>App</c>:
+	/// Entry point for the room display application.  
+	/// This app is designed to run on a mobile device that is hung outside a of a conference room 
+	/// to display details about the current meeting.
+	/// </summary>
+	// configurable options
+	readonly int RefreshRoomDataTimeInterval = 5000; // in milliseconds should be 5 min
+	public static TimeZoneInfo easternZone = TZConvert.GetTimeZoneInfo("America/New_York");
+	#if RELEASE
+		public static String deviceName = DeviceInfo.Name;
+	#endif
+	#if DEBUG
+		public static String deviceName = "Meeting Room 1";
+	#endif
+
+	RoomDataService roomDataService = new RoomDataService(); // This should be injected rather than instantiated.  
+	public static List<Meeting> meetingList = new List<Meeting>(); // List of meetings for room defined by deviceName 
+	private static System.Timers.Timer dataRefreshTimer; // Timer object used to repeatedly retreive room data see dataRefreshTimerCallback for details.
+
+	// Various application pages 
+	ContentPage hasMeetings = new MainPage();
+	ContentPage hasNoMeetings = new NoMeetingsPage();
+	ContentPage loadingDataPage = new LoadingDataPage();
+
 	public App()
 	{
 		InitializeComponent();
-		getData();
-
-		MainPage = new MainPage();
+		MainPage = loadingDataPage;
+		ConfgiureUpdateDataTimer();
 	}
 
-	private void getData()
+	private void ConfgiureUpdateDataTimer()
     {
+		// Create a timer and set a two second interval.
+		dataRefreshTimer = new System.Timers.Timer();
+		dataRefreshTimer.Interval = RefreshRoomDataTimeInterval;
+		// Hook up the Elapsed event for the timer. 
+		dataRefreshTimer.Elapsed += dataRefreshTimerCallback;
+		// Have the timer fire repeated events (true is the default)
+		dataRefreshTimer.AutoReset = true;
+		// Start the timer
+		dataRefreshTimer.Enabled = true;
+	}
+
+	private void dataRefreshTimerCallback(Object source, System.Timers.ElapsedEventArgs e)
+	{
+		///<summary>
+		/// Timmer callback. Repsonsible for getting room data, post procssing, and setting active page display. 
+		/// </summary>
 		meetingList = roomDataService.GetRoomData();
+		trimRoomDataByDeviceName();
+		setPage();
+		System.Diagnostics.Debug.WriteLine(roomDataService.lastUpdated);
+	}
+
+	private void trimRoomDataByDeviceName()
```

Comment:
Why is this a separate function from **dataRefreshTimerCallback**? Will there be a unit test for this, maybe?

Reply:
It keeps that method smaller, and yes it also helps to write tests.  I'm sure you have noticed we don't have any tests yet.  It's something I haven't figured out yet. By that I mean, how to write unit tests, how to mock data,  and how to run unit tests automatically.